spec/facebook_ads/model_spec.rb

```diff
@@ -0,0 +1,220 @@
+      expect(client)
+        .to receive(:get).with(anything, params: { fields: fetch_fields.first })
+      subject
+    end
+
+    describe 'deserialization' do
+      let(:foo_value) { 'A string' }
+      let(:response) { { 'foo' => foo_value } }
+
+      it 'populates the value of foo' do
+        subject
+        expect(test_model.foo).to eql foo_value
+      end
+
+      # Not going to be exhaustive here
+      # but at least check one other type than String
+      context 'value is an integer' do
+        let(:foo_value) { 1337 }
+
+        it 'populates the value of foo' do
+          subject
+          expect(test_model.foo).to eql foo_value
+        end
+      end
+
+      describe 'custom field types' do
+        let(:converted_value) { double('converted value') }
+        let(:converter) { double('Converter', from_json: converted_value) }
+        let(:test_class) do
+          Class.new(described_class) do
+            field :foo, type: :bar
+          end
+        end
+
+        before do
+          allow(FacebookAds::JsonConverter)
+            .to receive(:for).and_return(converter)
+        end
+
+        it 'acquires the correct kind of converter' do
+          expect(FacebookAds::JsonConverter)
+            .to receive(:for).with(:bar).and_return(converter)
+          subject
+        end
+
+        it 'populates the value of foo with the converted value' do
+          subject
+          expect(test_model.foo).to eql converted_value
+        end
+      end
+
+      context 'response contains unknown fields' do
+        let(:response) { { 'foo' => foo_value, 'unknown' => 'baz' } }
+
+        it 'populates the value of foo' do
+          subject
+          expect(test_model.foo).to eql foo_value
+        end
+      end
+    end
+
+    context 'calling without an array' do
+      subject { test_model.fetch(fetch_fields.first) }
+
+      it 'queries the specified field from the object' do
+        expect(client)
+          .to receive(:get)
+          .with(anything, params: { fields: fetch_fields.first })
+        subject
+      end
+    end
+
+    context 'requesting multiple fields' do
+      let(:fetch_fields) { %w(foo bar) }
+
+      it 'correctly formats multiple fields' do
+        expected_fields = 'foo,bar'
+        expect(client)
+          .to receive(:get).with(anything, params: { fields: expected_fields })
+        subject
+      end
+    end
+
+    context 'invalid field' do
+      let(:fetch_fields) { ['invalid'] }
+      it 'raises an error' do
+        expect { subject }.to raise_error(ArgumentError)
+      end
+    end
+  end
+
+  describe '#push' do
+    subject { test_model.push }
+    let(:response) { {} }
+
+    before do
+      allow(client).to receive(:post).and_return(response)
+    end
+
+    it 'pushes the data to the correct object' do
+      expect(client).to receive(:post).with(id, anything)
+      subject
+    end
+
+    it 'sends no fields' do
```

Comment:
The context should be called `'when no fields are known'`, but I agree.

(you never call `push` with parameters)